### ISSUE TITLE
ASDF tests change state of equivalencies in units

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#10283]
I haven't yet tracked down the origin of this, but if ``asdf`` is installed, the following fails:

```
$ pip install -e .
...
$ pytest astropy/coordinates/tests/test_angles.py astropy/io/misc/asdf/tags/unit/tests/test_equivalency.py astropy/modeling/tests/test_quantities_evaluation.py --sw
====================================================================== test session starts ======================================================================
platform linux -- Python 3.8.2, pytest-5.3.5, py-1.8.1, pluggy-0.13.1

Running tests with Astropy version 4.1.dev1579+g770bd2c5c.d20200503.
Running tests in astropy/coordinates/tests/test_angles.py astropy/io/misc/asdf/tags/unit/tests/test_equivalency.py astropy/modeling/tests/test_quantities_evaluation.py.

Date: 2020-05-03T17:36:40

Platform: Linux-5.4.0-26-generic-x86_64-with-glibc2.29

Executable: /home/tom/python/dev/bin/python3.8

Full Python Version: 
3.8.2 (default, Apr 27 2020, 15:53:34) 
[GCC 9.3.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.16.6
Scipy: 1.4.1
Matplotlib: 3.2.0
h5py: 2.10.0
Pandas: 1.0.1
Cython: not available
Scikit-image: 0.16.2
asdf: 2.6.0

Using Astropy options: remote_data: none.

Matplotlib: 3.2.0
Freetype: 2.6.1
rootdir: /home/tom/Code/astropy, inifile: setup.cfg
plugins: openfiles-0.4.1.dev10+g0203b1b, astropy-header-0.1.2, asdf-2.6.0, arraydiff-0.3, remotedata-0.3.2, mpl-0.11, filter-subpackage-0.1.1, cov-2.8.1, doctestplus-0.5.0, hypothesis-5.8.0
collected 89 items                                                                                                                                              
stepwise: no previously failed tests, not skipping.

astropy/coordinates/tests/test_angles.py ............................................                                                                     [ 49%]
astropy/io/misc/asdf/tags/unit/tests/test_equivalency.py .........................                                                                        [ 77%]
astropy/modeling/tests/test_quantities_evaluation.py .F

=========================================================================== FAILURES ============================================================================
________________________________________________________ test_evaluate_with_quantities_and_equivalencies ________________________________________________________

    def test_evaluate_with_quantities_and_equivalencies():
        """
        We now make sure that equivalencies are correctly taken into account
        """
    
        g = Gaussian1D(1 * u.Jy, 10 * u.nm, 2 * u.nm)
    
        # We aren't setting the equivalencies, so this won't work
        with pytest.raises(UnitsError) as exc:
>           g(30 * u.PHz)
E           Failed: DID NOT RAISE <class 'astropy.units.core.UnitsError'>

astropy/modeling/tests/test_quantities_evaluation.py:77: Failed
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: Test failed, continuing from this test next run. !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================= 1 failed, 70 passed in 4.76s ==================================================================

```

(The inclusion of a coordinates test file is to avoid https://github.com/astropy/astropy/issues/10280 by triggering an astropy.coordinates import)

It looks like the ASDF test for units is causing a permanent change in the equivalencies state. I'm not 100% sure why this hasn't shown up in CI though :thinking: 

(EDIT: this happens only if specutils is installed)

**Comment by astrofrog** [comment]
This appears to happen if specutils is installed, perhaps because specutils currently also defines an ASDF schema for SpectralCoord? (cc @eteq @nmearl @jdavies-st)

**Comment by mhvk** [comment]
Just in case it helps, look for `{add|set}_enabled_equivalencies` - good to do this with context managers, or have a `setup_class/teardown_class` pair that ensures things are left in their original state.

**Comment by astrofrog** [comment]
Indeed! https://github.com/astropy/specutils/blob/ca093ea8757d17a728f1ee1adafc04bbeff083c4/specutils/spectra/spectrum1d.py#L18

**Comment by astrofrog** [comment]
Issue opened in https://github.com/astropy/specutils/issues/671

**Comment by dhomeier** [comment]
Should be closed with astropy/specutils#673.
